share/ci/install/requirements_txt_modifier.py

```diff
@@ -53,21 +53,27 @@ def cs(text: str, color: str):
 
 if __name__ == "__main__":
     if len(sys.argv) < 3:
-        print("Set path to requirements.txt as first argument and output path as" " second argument.")
-        exit(1)
+        print("Set the path of the input requirements.txt as the second last argument "
+              "and the path of the output requirements.txt as the last argument.")
+        print("Set the name of the environment variables to ignore "
+              "before the path of the input requirements.txt")
+        sys.exit(1)
+
+    env_to_ignore = sys.argv[1:-2]
 
     # parse environment variables
     packages = {}
     for envvar in os.environ:
-        if envvar.startswith("PYPIC_DEP_VERSION_"):
-            packages[envvar.split("_")[-1]] = os.environ[envvar]
+        if not envvar in env_to_ignore:
+            if envvar.startswith("PYPIC_DEP_VERSION_"):
+                packages[envvar.split("_")[-1]] = os.environ[envvar]
 
     print("Try to set following package to a fix version")
     for pkg_name, pkg_version in packages.items():
         print(f"  {pkg_name} -> {pkg_version}")
 
-    with open(sys.argv[1], "r", encoding="utf-8") as input:
-        with open(sys.argv[2], "w", encoding="utf-8") as output:
+    with open(sys.argv[-2], "r", encoding="utf-8") as input:
+        with open(sys.argv[-1], "w", encoding="utf-8") as output:
```

Comment:
This looks horribly dangerous. I'm aware that we're guarding against less than 3 arguments above but if the next person touching this file isn't aware of that, we might write the content of this very Python script into the output file. Maybe we could already extract the filenames in the else branch of the `if len(...` condition above, so there is at least proximity between those.

Reply:
If I want make it save, I need to use `argparse`. I check the option.

Reply:
As I said, an intermediate step towards safety could be to extract those in the `if` condition.